### PR TITLE
fix(alerting): MatchType and TimeIntervalItem

### PR DESCRIPTION
**Pull request by Duologic**
Port fixes from Grafonnet to here.

### CODE REVIEW

File: config/compiler/alerting.yaml [hunk 1]
```diff
@@ -55,3 +55,22 @@ passes:
...
+          value_type:
+            kind: ref
+            ref: { referred_pkg: alerting, referred_type: TimeIntervalItem }
+  - retype_object:
+      object: alerting.MatchType
+      as:
+        kind: enum
+        enum:
+          values:
+            - { type: { kind: scalar, scalar: { scalar_kind: string } }, value: '=' }
```
Comment (K-Phoen) [inline config/compiler/alerting.yaml:72]: Enum members have names :)

https://github.com/grafana/cog/blob/56dd13d47a04682cee1c170c950f6862b4903222/internal/ast/types.go#L651

Comment (Duologic) [reply config/compiler/alerting.yaml:72]: how and where are they used?

Comment (K-Phoen) [reply config/compiler/alerting.yaml:72]: In jennies, to generate stuff like: https://github.com/grafana/cog/blob/56dd13d47a04682cee1c170c950f6862b4903222/testdata/jennies/rawtypes/enums/GoRawTypes/enums/types_gen.go#L6

**Review by K-Phoen — APPROVED** [review]
LGTM :rocket:

**Pull request merged**